Makefile

```diff
@@ -43,98 +42,73 @@ DEV_MOUNTS = '-v $(CWD)/$(FV3):/$(FV3)/$(FV3) -v $(CWD)/tests:/$(FV3)/tests -v $
 
-build_environment:
-	DOCKER_BUILDKIT=1 docker build \
-		--network host \
-		--build-arg MIDBASE=$(BASE_INSTALL) \
-		-f $(CWD)/docker/Dockerfile.build_environment \
-		-t $(FV3_INSTALL_IMAGE) \
-		--target $(FV3_INSTALL_TARGET) \
-		.
-
-build_wrapped_environment:
-	DOCKER_BUILDKIT=1 docker build \
-		--network host \
-		-f $(CWD)/docker/Dockerfile.build_environment \
-		-t $(WRAPPER_INSTALL_IMAGE) \
-		--target $(FV3_INSTALL_TARGET) \
-		--build-arg BASE_IMAGE=$(WRAPPER_IMAGE) \
-		.
+# Image build instructions have moved to docker/Makefile but are kept here for backwards-compatibility
+
+build_environment: update_submodules
+	$(MAKE) -C docker build_core_deps
+
+build_wrapped_environment: update_submodules
+	$(MAKE) -C docker build_deps
 
 build: update_submodules
 	if [ $(PULL) == True ]; then \
 		$(MAKE) pull_environment_if_needed; \
 	else \
 		$(MAKE) build_environment; \
 	fi
-	docker build \
-		--network host \
-		--build-arg build_image=$(FV3_INSTALL_IMAGE) \
-		-f $(CWD)/docker/Dockerfile \
-		-t $(FV3_IMAGE) \
-		.
+	$(MAKE) -C docker fv3core_image
 
 build_wrapped: update_submodules
 	if [ $(PULL) == True ]; then \
 		$(MAKE) pull_wrapped_environment_if_needed; \
 	else \
-		$(MAKE) build_environment; \
-	$(MAKE) build_wrapped_environment
-	docker build \
-		--network host \
-		--no-cache \
-		--build-arg build_image=$(WRAPPER_INSTALL_IMAGE) \
-		-f $(CWD)/docker/Dockerfile \
-		-t $(WRAPPED_FV3_IMAGE) \
-		.
-
+		$(MAKE) build_wrapped_environment; \
+  fi
+  $(MAKE) -C docker fv3core_wrapper_image
 
 pull_environment_if_needed:
-	if [ -z $(shell docker images -q $(FV3_INSTALL_IMAGE)) ]; then \
-		docker pull $(FV3_INSTALL_IMAGE); \
-	fi
+	$(MAKE) -C docker pull_core_deps_if_needed
 
 pull_wrapped_environment_if_needed:
 	if [ -z $(shell docker images -q $(WRAPPER_INSTALL_IMAGE)) ]; then \
 		docker pull $(WRAPPER_INSTALL_IMAGE); \
 	fi
 
 pull_environment:
-	docker pull $(FV3_INSTALL_IMAGE)
+	$(MAKE) -C docker pull_core_deps
 
 push_environment:
-	docker push $(FV3_INSTALL_IMAGE)
+	$(MAKE) -C docker push_deps
 
-rebuild_environment: build_environment
-	$(MAKE) push_environment
+rebuild_environment: build_environment push_environment
 
 push_core:
-	docker push $(FV3_IMAGE)
+	$(MAKE) -C docker push
 
 pull_core:
-	docker pull $(FV3_IMAGE)
+	$(MAKE) -C docker pull
 
 tar_core:
-	docker save $(FV3_IMAGE) -o $(CORE_TAR)
-	gsutil copy $(CORE_TAR) $(CORE_BUCKET_LOC)
+	$(MAKE) -C docker tar_core
 
 sarus_load_tar:
-	export FOUND_IMAGE=`sarus images | grep $(FV3_TAG)`
-	if [ -z $(FOUND_IMAGE) ] && [ ! -f `pwd`/$(CORE_TAR) ]; then \
-		gsutil copy $(CORE_BUCKET_LOC) . && \
-		sarus load ./$(CORE_TAR) $(FV3_TAG); \
-        fi
+	$(MAKE) -C docker sarus_load_tar
 
 cleanup_remote:
-	gsutil rm $(CORE_BUCKET_LOC)
-	gcloud container images delete -q --force-delete-tags $(FV3_IMAGE)
+	$(MAKE) -C docker cleanup_remote
+
+# end of image build targets which have been moved to docker/Makefile
 
 tests: build
```

Comment:
Not sure what happened, but now using this Makefile gives an error:
 *** missing separator.  Stop
Must be a whitespace complaint of some sort?

Reply:
Thanks! Don't know how I didn't see this.